gtk/main.cc

```diff
@@ -127,12 +128,31 @@ int main(int argc, char** argv)
         config_dir = tr_getDefaultConfigDir(AppConfigDirName);
     }
 
+    auto lock = tr_get_lock_name(config_dir);
+    if (std::filesystem::exists(lock))
+    {
+        fmt::print(stderr, "Detected another instance using the same configuration directory\n");
+        fmt::print(stderr, "Remove the file \"{}\" if you are sure this is a false error\n", lock.c_str());
+        return 2;
+    }
+    else
+    {
+        std::ofstream f(lock);
```

Comment:
Race condition between checking for lock file existence and its creation. Since multiple processes are usually allowed to open the same file for writing, no error will occur as long as the file didn't exist when they checked for it concurrently. 

Existing file-based locking could be found in libtransmission/session-id.{h,cc}. In fact, I'd be more open to just using it instead, and moving configuration directory locking to `tr_sessionInit()`, with clients being ready to handle the potential initialization failure.